### ISSUE TITLE
Problem with remoteDriver()

### ISSUE THREAD

**Issue by kferris10** [ropensci/RSelenium#49]
I am trying to use `remoteDriver(browserName = "firefox")` and am running into an error.

    > library(RSelenium)
    > checkForServer()
    > startServer()
    Warning message:
    running command '"java" -jar "C:/Users/chessa015/Documents/R/win-library/3.1/RSelenium/bin/selenium-server-standalone.jar" -log "C:/Users/chessa015/Documents/R/win-library/3.1/RSelenium/bin/sellog.txt"' had status 127  
    > Sys.sleep(15)
    > remDr <- remoteDriver(browserName = "firefox")
    > remDr$open()
    [1] "Connecting to remote server"
    Undefined error in RCurl call.
    Error in queryRD(paste0(serverURL, "/session"), "POST", qdata = toJSON(serverOpts)) : 

I encountered this issue in tdhock/animint#66.  Please let me know what else I can do to help.

    > devtools::session_info()
    Session info ---------------------------------------------------------------
     setting  value                       
     version  R version 3.1.3 (2015-03-09)
     system   x86_64, mingw32             
     ui       RStudio (0.98.1083)         
     language (EN)                        
     collate  English_United States.1252  
     tz       America/Denver              
    
    Packages -------------------------------------------------------------------
     package    * version   date       source        
     bitops       1.0-6     2013-08-17 CRAN (R 3.1.1)
     caTools    * 1.17.1    2014-09-10 CRAN (R 3.1.1)
     devtools   * 1.7.0     2015-01-17 CRAN (R 3.1.2)
     RCurl        1.95-4.5  2014-12-06 CRAN (R 3.1.2)
     RJSONIO      1.3-0     2014-07-28 CRAN (R 3.1.1)
     RSelenium    1.3.5     2014-10-26 CRAN (R 3.1.3)
     rstudio    * 0.98.1083 2014-10-26 local         
     rstudioapi * 0.2       2014-12-31 CRAN (R 3.1.2)
     XML          3.98-1.1  2013-06-20 CRAN (R 3.1.1)

**Comment by johndharrison** [comment]
Hi Kevin,

Do you have a selenium server present in RSelenium/bin folder. You can run:

```
RSelenium::checkForServer()
```
that should download the latest selenium server to RSelenium/bin. Otherwise you may not have java installed on your windows box.

```
`> Sys.which("java")
                                            java 
"C:\\PROGRA~3\\Oracle\\Java\\javapath\\java.exe" 
``

**Comment by kferris10** [comment]
I did not have java installed.  Thanks for your help!

**Comment by Sayantan2015** [comment]
I have updated Java. Still I am getting an error . when I am trying to run this code : 

remDr <- remoteDriver$new()
remDr$open()

[1] "Connecting to remote server"
Error: 	 Summary: UnknownError
 	 Detail: An unknown server-side error occurred while processing the command.
 	 class: org.openqa.selenium.WebDriverException

**Comment by cmohamma** [comment]
I am also getting a similar error. I'm trying to run this on Yosemite. I tried it with newer and older versions of Java but both give me the following error:

Undefined error in RCurl call.Error in queryRD(paste0(serverURL, "/session"), "POST", qdata = toJSON(serverOpts)) : 

Any progress on this issue?

**Comment by johndharrison** [comment]
Can you try with the most recent dev version:

```
devtools::install_github("ropensci/RSelenium")
```
And report any errors

**Comment by cmohamma** [comment]
Thanks for the suggestion, I installed from github. This then produced a new error. If I toggled the remote driver to google then I get the following new error:

Selenium message: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html 
Error: 	 Summary: UnknownError
 	 Detail: An unknown server-side error occurred while processing the command.
 	 class: java.lang.IllegalStateException
	 Further Details: run errorDetails method

If I set the driver to firefox I get the following:

Selenium message: Unsupported Marionette protocol version 2, required 3 (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 966 milliseconds
Build info: version: '3.0.0-beta2', revision: '2aa21c1', time: '2016-08-02 15:03:28 -0700'
System info: host: 'Cyrus-M', ip: '192.168.1.71', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.11.5', java.version: '1.8.0_91'
Driver info: org.openqa.selenium.firefox.FirefoxDriver 
Error: 	 Summary: UnknownError
 	 Detail: An unknown server-side error occurred while processing the command.
 	 class: org.openqa.selenium.WebDriverException
	 Further Details: run errorDetails method

Not sure what's going on here but on stack overflow there are at least three unanswered questions in regards to individuals trying to implement Rselenium on OS X El Capitan. It seems to be a problem related only to those on Macs with the newest OS.

**Comment by johndharrison** [comment]
@cmohamma You can start the selenium server giving it the path to the chromedriver:

```
RSelenium::startServer(javaargs = c("-Dwebdriver.chrome.driver=\"pathTOdriver/****.exe\""))
```
Mozilla Firefox now requires the gecko driver and marionette see #81

**Comment by cmohamma** [comment]
@johndharrison thanks, how would that look on a mac? I tried ``RSelenium::startServer(javaargs = c("-Dwebdriver.chrome.driver="/users/myname/chromedriver""))``  And I get the following error: Error: unexpected string constant in "RSelenium::startServer(javaargs = c("-Dwebdriver.chrome.driver="/users/cyrusmohammadian/chromedriver.bin"""

**Comment by johndharrison** [comment]
You to escape the quotation marks. Or:

```
RSelenium::startServer(javaargs = c('-Dwebdriver.chrome.driver="pathTOdriver/****.exe"))
```

**Comment by cmohamma** [comment]
right, got it! thanks!

**Comment by cmohamma** [comment]
Everything worked fine for a couple days but now I'm encountering this error!! I cannot believe it. It worked fine for TWO DAYS! 

Undefined error in RCurl call.
Error in queryRD(paste0(serverURL, "/session"), "POST", qdata = toJSON(serverOpts)) :

**Comment by johndharrison** [comment]
Your browser or driver has changed. If you continue having problems with running a Selenium Server you can consider using Docker containers:

http://rpubs.com/johndharrison/RSelenium-Docker

**Comment by cmohamma** [comment]
Thanks for the response! All good now!

**Comment by MesiasRaul** [comment]
Hi, currently what is the most updated version of RSelenium?

When I tried to use:

remoteDriver(browserName = "chrome")$open I received:

[1] "Connecting to remote server"
Error in checkError(res) : 
  Undefined error in httr call. httr output: Send failure: Connection was reset

And when I tried:

RSelenium::startServer(javaargs = c('-Dwebdriver.chrome.driver=\"pathTOdriver/****.exe\"'))

Error in RSelenium::startServer(javaargs = c("-Dwebdriver.chrome.driver=\"pathTOdriver/****.exe\"")) : 
  No Selenium Server binary exists. Run checkForServer or start 
           server manually.
In addition: Warning message:
startServer is deprecated.
Users in future can find the function in 
file.path(find.package("RSelenium"), "example/serverUtils").
The sourcing/starting of a Selenium Server is a users responsiblity. 
Options include manually starting a server see 
vignette("RSelenium-basics", package = "RSelenium")
and running a docker container see  
vignette("RSelenium-docker", package = "RSelenium")

Sorry for my english.
